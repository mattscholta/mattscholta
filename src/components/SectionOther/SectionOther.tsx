import * as React from 'react';
import classnames from 'classnames';

import { data as images } from '@src/data/images';

import styles from './styles.module.css';

export interface SectionOtherProps {
  className?: string;
}

/**
 * @name SectionOther
 * @description Stateless Component / Pure Function
 */
const SectionOther: React.FC<SectionOtherProps> = (props) => {
  const { className } = props;

  // Styles
  const classesImage = classnames('shadow', styles.image);

  // Markup
  const renderImage = (src: string) => (
    <img
      alt=""
      className={classesImage}
      key={src}
      loading="lazy"
      src={src}
      width={120}
    />
  );

  return (
    <section className={className}>
      <h2 className="ui-heading">Free Time</h2>
      <span className="ui-subhead" />

      <p>
        A true <b>California native</b> you can find me and the family at the
        beach! When we're not at the beach there is a good chance I am hauling
        ass along the coast on my motorcycle üèçÔ∏èüí®.
      </p>

      <div className={styles.images}>{images.map(renderImage)}</div>
    </section>
  );
};

export { SectionOther as default, SectionOther };
