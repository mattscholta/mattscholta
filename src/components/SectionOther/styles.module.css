.images {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin-bottom: var(--grid-8x);
}

.image {
  flex: 1 0;
  margin: var(--grid-2x);
}

@media (max-width: 768px) {
  .images {
    display: block;
  }

  .image {
    margin: 2%;
    width: 29.33%;
  }
}

@media (max-width: 550px) {
  .images {
    display: block;
  }

  .image {
    width: 46%;
  }
}
