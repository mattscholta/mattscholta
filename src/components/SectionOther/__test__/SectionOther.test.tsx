import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { SectionOther, SectionOtherProps } from '../SectionOther';

describe('SectionOther', () => {
  let props: SectionOtherProps;
  let wrapper: RenderResult;

  const createWrapper = () => <SectionOther {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
