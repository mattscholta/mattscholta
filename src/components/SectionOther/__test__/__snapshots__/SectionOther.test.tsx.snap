// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionOther matches snapshot 1`] = `
NodeList [
  <section>
    <h2
      class="ui-heading"
    >
      Free Time
    </h2>
    <span
      class="ui-subhead"
    />
    <p>
      A true 
      <b>
        California native
      </b>
       you can find me and the family at the beach! When we're not at the beach there is a good chance I am hauling ass along the coast on my motorcycle üèçÔ∏èüí®.
    </p>
    <div
      class="images"
    >
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/04.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/21.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/05.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/22.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/12.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/23.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/13.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/17.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/10.jpg"
        width="120"
      />
      <img
        alt=""
        class="shadow image"
        loading="lazy"
        src="https://s3-us-west-1.amazonaws.com/mattscholta/instagram/06.jpg"
        width="120"
      />
    </div>
  </section>,
]
`;
